include/DataFile.h

```diff
@@ -58,6 +58,8 @@ class EXPORT DataFile : public QDomDocument
 
 	virtual ~DataFile();
 
+	bool validate( QString extension );
```

Comment:
Don't we already have this information when the object is instantiated via the full name?

I guess we don't really need this argument, unless you want to split between instance and static methods.